tests/pipeline/snapshots/snap_test_example.py

```diff
@@ -210,6 +210,7 @@
                 },
             },
             "from": {
+                "components": [],
```

Comment:
Should it be excluded from the generated YAML, or is it fine to have it as an empty list?

Reply:
Can they also both exist? concretely, reading from topic(s) and component(s) at the same time

Reply:
I can't think of any downsides other than it being a bit verbose maybe. that being said I don't have a preference, fine with either option 🤷🏻

Reply:
updated the test & snapshot